src/helper.js

```diff
@@ -138,6 +138,18 @@ const escapeRegex = (str) => {
   return str.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&')
 }
 
+/**
+ * @param {string} line
+ * @return {number}
+ */
+const parseLogDate = (line) => {
+  const logDate = Date.parse(line.split(' ')[0])
+
+  if (Number.isNaN(logDate)) throw new Error('the log line don\'t have a timestamp')
```

Comment:
hmm consider this scenarion:

**log content**
```
2022-01-17T15:26:18.863Z Error: foo
    at REPL5:1:41
    at Script.runInThisContext (vm.js:134:12)
    at REPLServer.defaultEval (repl.js:566:29)
    at bound (domain.js:421:15)
    at REPLServer.runBound [as eval] (domain.js:432:12)
    at REPLServer.onLine (repl.js:909:10)
    at REPLServer.emit (events.js:412:35)
    at REPLServer.emit (domain.js:475:12)
    at REPLServer.Interface._onLine (readline.js:434:10)
    at REPLServer.Interface._line (readline.js:791:8)
undefined
2022-01-18T15:26:19.835Z Error: foo
    at REPL6:1:41
    at Script.runInThisContext (vm.js:134:12)
    at REPLServer.defaultEval (repl.js:566:29)
    at bound (domain.js:421:15)
    at REPLServer.runBound [as eval] (domain.js:432:12)
    at REPLServer.onLine (repl.js:909:10)
    at REPLServer.emit (events.js:412:35)
    at REPLServer.emit (domain.js:475:12)
    at REPLServer.Interface._onLine (readline.js:434:10)
    at REPLServer.Interface._line (readline.js:791:8)
undefined
2022-01-19T15:26:20.746Z Error: foo
    at REPL7:1:41
    at Script.runInThisContext (vm.js:134:12)
    at REPLServer.defaultEval (repl.js:566:29)
    at bound (domain.js:421:15)
    at REPLServer.runBound [as eval] (domain.js:432:12)
    at REPLServer.onLine (repl.js:909:10)
    at REPLServer.emit (events.js:412:35)
    at REPLServer.emit (domain.js:475:12)
    at REPLServer.Interface._onLine (readline.js:434:10)
    at REPLServer.Interface._line (readline.js:791:8)
undefined
2022-01-19T15:26:21.394Z Error: foo
    at REPL8:1:41
    at Script.runInThisContext (vm.js:134:12)
    at REPLServer.defaultEval (repl.js:566:29)
    at bound (domain.js:421:15)
    at REPLServer.runBound [as eval] (domain.js:432:12)
    at REPLServer.onLine (repl.js:909:10)
    at REPLServer.emit (events.js:412:35)
    at REPLServer.emit (domain.js:475:12)
    at REPLServer.Interface._onLine (readline.js:434:10)
    at REPLServer.Interface._line (readline.js:791:8)
undefined
```

**writer**
```
node bin/hyperlog.js write --file ../logs/log-1.txt --republish
  hcore-logger key: 1c54bcb92a687004b04f364997162c4482356a14ec57653b36ea4834f8b46d60 +0ms
  hcore-logger secret-key: b1663dce53c05feb571dacc83c50db3851d65b90310201dc90d41dd60c6bdb751c54bcb92a687004b04f364997162c4482356a14ec57653b36ea4834f8b46d60 +0ms
  hcore-logger feed started listening for changes on ../logs/log-1.txt +0ms
```

**reader**
```
node bin/hyperlog.js read -k 1c54bcb92a687004b04f364997162c4482356a14ec57653b36ea4834f8b46d60 --start-date '2022-01-19 15:26:20'
***/hypercore-logs/src/helper.js:148
  if (Number.isNaN(logDate)) throw new Error('the log line don\'t have a timestamp')
                             ^

Error: the log line don't have a timestamp
    at parseLogDate (***/hypercore-logs/src/helper.js:148:36)
```

We should find a way to handle these logs, maybe we can store prev timestamp and return that one in case of error? default previous time stamp could be unix epoch start (1970)

Reply:
https://github.com/bitfinexcom/hypercore-logs/pull/11/commits/1ae86f960b5cc5cc118dc7a2fbe1bd1f75755447